examples/scripts/blender/sdf_exporter.py

```diff
@@ -0,0 +1,206 @@
+import bpy
+import os.path
+from bpy_extras.io_utils import ImportHelper
+from bpy.props import StringProperty, BoolProperty
+from bpy.types import Operator
+
+import xml.etree.ElementTree as ET
+from xml.dom import minidom
+
+# Target blender version: 2.82
+
+########################################################################################################################
+### Exports model.dae of the scene with textures, its corresponding model.sdf file, and a default model.config file ####
+########################################################################################################################
+def export_sdf(prefix_path):
+
+    dae_filename = 'model.dae'
+    sdf_filename = 'model.sdf'
+    model_config_filename = 'model.config'
+    lightmap_filename = 'LightmapBaked.png'
+    
+    # Exports the dae file and its associated textures
+    bpy.ops.wm.collada_export(filepath=prefix_path+dae_filename, check_existing=False, filter_blender=False, filter_image=False, filter_movie=False, filter_python=False, filter_font=False, filter_sound=False, filter_text=False, filter_btx=False, filter_collada=True, filter_folder=True, filemode=8)
+
+    # objects = bpy.context.selected_objects
+    objects = bpy.context.selectable_objects
+    mesh_objects = [ o for o in objects if o.type == 'MESH' ]
+    light_objects = [ o for o in objects if o.type == 'LIGHT' ]
+
+    #############################################
+    #### export sdf xml based off the scene #####
+    #############################################
+    sdf = ET.Element('sdf', attrib={'version':'1.8'})
+
+    # 1 model and 1 link
+    model = ET.SubElement(sdf, "model", attrib={"name":"test"})
+    static = ET.SubElement(sdf, "static")
+    static.text = "true"
+    link = ET.SubElement(model, "link", attrib={"name":"testlink"})
+    # for each geometry in geometry library add a <visual> tag
+    for o in mesh_objects:
+        visual = ET.SubElement(link, "visual", attrib={"name":o.name})
+
+        geometry = ET.SubElement(visual, "geometry")
+        mesh = ET.SubElement(geometry, "mesh")
+        uri = ET.SubElement(mesh, "uri")
+        uri.text = dae_filename
```

Comment:
It's common practice to place the meshes in a `meshes` directory and the textures in `materials/textures`. Would it be too troublesome to generate that directory structure?

Reply:
Moved the dae export to the meshes folder d6222f9

There is a problem that the textures have to be exported at the same location as the dae. At the moment the fix would entail lifting the entire collada export function and modifying it, plus the relocation of the texture files. I wouldn't do it within this script.

This is something I'll fix at the import stage, specifically for the collada world exporter, which places textures at the same location of the dae file.

Reply:
Got it, it's ok like that. It's more of a convention than a rule, Ignition supports any folder structure if the paths are declared correctly.